nav2_route/src/route_planner.cpp

```diff
@@ -161,7 +164,9 @@ bool RoutePlanner::getTraversalCost(
     return true;
   }
 
-  return edge_scorer_->score(edge, goal, isGoal(edge->end), score);
+  enum EdgeType edge_type = classifyEdge(edge);
+
+  return edge_scorer_->score(edge, goal, edge_type, score);
```

Comment:
```suggestion
  return edge_scorer_->score(edge, goal, classifyEdge(edge), score);
```

And remove the previous line, its more concise and I think still very clear